app/public/cantusdata/helpers/mei_processing/mei_parsing_types.py

```diff
@@ -0,0 +1,94 @@
+"""
+Contains type definitions used in the MEI parsing process.
+"""
+
+from typing import Tuple, TypedDict, Literal, List, Optional
+from typing_extensions import TypeAlias
+
+# A type for coordinates of bounding boxes
+CoordinatesType: TypeAlias = Tuple[int, int, int, int]
+"""
+A type for coordinates of bounding boxes. The coordinates
+of the box are given as four integers designating, in order:
+    - the x-coordinate of the upper-left corner of the box
+    - the y-coordinate of the upper-left corner of the box
+    - the x-coordinate of the lower-right corner of the box
+    - the y-coordinate of the lower-right corner of the box
+"""
+
+
+class Zone(TypedDict):
+    """A type for zones (bounding boxes) in MEI files.
+
+    coordinates: The location of the bounding box as
+        defined in MEI 'zone' elements as a tuple of
+        type CoordinatesType.
+    rotate: The rotation of the zone in degrees.
+    """
+
+    coordinates: CoordinatesType
+    rotate: float
+
+
+ContourType = Literal["u", "d", "s"]
```

Comment:
I'm not sure a `PitchName` type is useful because we are extracting the pitch names from the xml document itself, so static type checking won't really be able to do anything with that. What we could do is run a check during the parsing that the pitch name that is extracted is valid (i.e. is one of "c", "d", "e", etc.). I guess that would be better than getting a random key error at https://github.com/DDMAL/cantus/blob/a62f5dd53d6f579556d40085118540b079ecacca/app/public/cantusdata/helpers/mei_processing/mei_parser.py#L368-L369

`NeumeType` is now added.

Reply:
Ok, I've added a `try`/`except` that raises a more descriptive error in this case.